pytorch_lightning/metrics/converters.py

```diff
@@ -241,6 +247,28 @@ def _tensor_collection_metric_conversion(func_to_decorate: Callable) -> Callable
     return _tensor_collection_metric_output_conversion(func_convert_inputs)
 
 
+def sync_dist_if_available(
+    result: Union[torch.Tensor], group: Optional[Any] = None, reduce_op: Optional[Union[ReduceOp, str]] = None
+) -> torch.Tensor:
+    """
+    Function to reduce a tensor across worker processes during distributed training
+
+    Args:
+        result: the value to sync and reduce (typically tensor or number)
+        group: the process group to gather results from. Defaults to all processes (world)
+        reduce_op: the reduction operation. Defaults to sum.
+            Can also be a string of 'avg', 'mean' to calculate the mean during reduction.
+
+    Return:
+        reduced value
+    """
+    if torch.distributed.is_available() and torch.distributed.is_initialized():
```

Comment:
so at firistround try PT distrib and later Horovod?

Reply:
At the moment, this component does not have access to the Trainer state the tells us which distributed backend we're using.  A better design would be to inject this somehow, but it's not immediately clear how we could do this, since the EvalResult is created by the LightningModule.